### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by NikitaKarbazin**

### CODE REVIEW

**Review by Kaunin1 — APPROVED** [review]

**Comment by Kaunin1** [comment]
Good job!

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,104 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
```
Comment (Olexandr6) [inline src/figuresClasses.ts:2]: Better to use `enum` for these values, cause you use them in your code as values

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,104 @@
...
 
+export interface Figure {
+  color:Color;
```
Comment (Olexandr6) [inline src/figuresClasses.ts:5]: Add a space for better readability
```suggestion
  color: Color;
```
```suggestion
  color:Color;
```

File: src/figuresClasses.ts [hunk 3]
```diff
@@ -1,19 +1,104 @@
...
+  c: number;
 
+  constructor(
+    color: Color,
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error();
```
Comment (Olexandr6) [inline src/figuresClasses.ts:28]: This error isn't informative for a user

File: src/figuresClasses.ts [hunk 4]
```diff
@@ -1,19 +1,104 @@
...
 
-}
+  shape:Shape = 'triangle';
+
+  a: number;
+
+  b: number;
 
-export class Circle {
+  c: number;
```
[the remark this answers is not in the document]
Comment (NikitaKarbazin) [reply src/figuresClasses.ts]: For me its better to understand!

File: src/figuresClasses.ts [hunk 5]
```diff
@@ -1,19 +1,90 @@
...
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(`Sides of ${this.shape} should be great than 0`);
+    }
+
+    if (Math.max(...[a, b, c]) >= a + b + c - Math.max(...[a, b, c])) {
+      throw new Error(`
+      For a ${this.shape}
+      , the sum of any two sides must be greater than the third.
+      Otherwise, the two sides will simply
+      "fall" on the third and the triangle will not work. FIX THIS!!!`);
```
Comment (Olexandr6) [inline src/figuresClasses.ts:33]: 😁

**Review by Olexandr6 — COMMENTED** [review]
Looks much better

**Review by Olexandr6 — APPROVED** [review]

**Pull request closed without merging**